src/bp/sdk/botpress.d.ts

```diff
@@ -2352,12 +2354,205 @@ declare module 'botpress/sdk' {
     export function getMessageSignature(message: string): Promise<string>
   }
 
+  export type uuid = string
+
   /**
    * These features are subject to change and should not be relied upon.
    * They will eventually be either removed or moved in another namespace
    */
   export namespace experimental {
     export function disableHook(hookName: string, hookType: string, moduleName?: string): Promise<boolean>
     export function enableHook(hookName: string, hookType: string, moduleName?: string): Promise<boolean>
+
+    export interface Conversation {
+      id: uuid
+      userId: string
+      botId: string
+      createdOn: Date
+    }
+
+    export interface RecentConversation extends Conversation {
+      lastMessage?: Message
+    }
+
+    export interface Message {
+      id: uuid
+      conversationId: uuid
+      eventId: string
+      incomingEventId: string
+      from: string
+      sentOn: Date
+      payload: any
+    }
+
+    export interface ListOptions {
+      limit?: number
+      offset?: number
+    }
```

Comment:
EventSearchParams - from/count

Reply:
limit and offset are pretty much standard syntax for most of SQL databases. NoSQL db such as Mongo or CouchDB use limit and skip. 
That being said, I don't mind let's just use the same thing all over the code base.

Reply:
Yeah I know and I agree ! But since we already have a different naming elsewhere, it makes sense to reuse it elsewhere. It's a better DX

Reply:
`EventSearchParams` has event in the name to begin with so we'd need to rename it a least. It also has options for sorting which messaging doesn't have on purpose. If you have millions of messages letting the user decide custom sorting might lead to really inefficient queries. 

Besides I don't see any reason that you would want to order conversations and messages by something else than their dates. If there are other sorting options that make sense, then they should probably be a whole other function, or provided as an hardcoded sorting option. I just don't like letting sdk users do whatever they want with sorting